### PR TITLE
Improve the visibility of the `.limit_toggler` in the back end

### PR DESCRIPTION
**Pull request by lukasbableck**
In Contao 5 if there is enough content in an element for its height to be limited, the content overlaps the .limit_toggler. This makes the toggler very hard to see. Also the content is cut off right at the border.
With this PR it looks more like Contao 4.13 again, but with a gradient at the bottom.

C4.13
image

C5.3 currently
image

C5.3 with this change
see https://github.com/contao/contao/pull/7367#issuecomment-2251526836

### CODE REVIEW

**Comment by bytehead** [comment]
Nice one, I've noticed this myself but didn't investigate 🙃

**Comment by zoglo** [comment]
I thought I fixed it in #6661

**Comment by lukasbableck** [comment]
No, https://github.com/contao/contao/pull/6661 only fixed that the toggler was not clickable

**Comment by asaage** [comment]
I still think it's pretty unnoticable https://github.com/contao/core-bundle/issues/1198

**Comment by zoglo** [comment]
How about adding this as well:

```diff
.limit_toggler {
	width: 100%;
	position: absolute;
	bottom: 0;
	left: 0;
-	background: var(--content-bg);
+	background: linear-gradient(transparent, var(--content-bg));
	line-height: 11px;
	text-align: center;
	z-index: 1;
}

.limit_toggler button {
	margin: 0;
	padding: 0;
-	border: 0;
+       border: 1px solid var(--border);
	background: var(--content-bg);
	width: 24px;
	line-height: 8px;
	color: var(--gray);
	border-top-left-radius: 2px;
	border-top-right-radius: 2px;
}
```

That one would make it more visible and make sure that people see there is more content as well <3

image
image

**Comment by leofeyer** [comment]
I agree that the gradient is the better solution. However, I cannot reproduce the initial problem:

**Comment by lukasbableck** [comment]
Apparantly Safari doesn't have that problem, only in Chrome, Firefox and everything based on them it looks like in my screenshot. Haven't noticed that until now.

**Comment by fritzmg** [comment]
I can reproduce the issue in Firefox.

**Comment by leofeyer** [comment]
Interesting. It seems that Firefox misses a `z-index: 1`.

**Comment by fritzmg** [comment]
The `order: 3` from `.cte_preview` changes the rendering stack so that `.cte_preview` is rendered _after_ `.limit_toggler` - thus putting the content of `.cte_preview` on top of `.limit_toggler`, which is somewhat expected.

**Comment by leofeyer** [comment]
I have added a separate PR #7374 that just fixes the missing `z-index`. Then we can keep this PR to discuss the gradient.





The problem with the second screenshot is that it still looks like it isn‘t fully expanded, although it is.

**Comment by lukasbableck** [comment]
Hmm, I can not reproduce that. Tried it in Safari, Firefox and Chrome on Contao 5.3.11.

But I have noticed that a background-color was set for the toggler on hover which looked weird as it was not transparent anymore. This is fixed now.

**Comment by zoglo** [comment]
@lukasbableck The gradient itself is always visible, however it may overlap and act weird in different browser engines.
We should definitely go with something very specific like this to fix it (feel free to add transitions):
```css
[data-contao--limit-height-target][style*="max-height"] .limit_toggler {
    background: linear-gradient(transparent, var(--content-bg));
}
```

**Comment by lukasbableck** [comment]
Thanks @zoglo, I have changed that.
Not sure about transitions in this case, gradients can't be easily transitioned anyway AFAIK. IMO it's ok like that.

**Comment by zoglo** [comment]
@lukasbableck there is one more issue that needs to be fixed

image
The toggler itself needs to be relative to the parent, however you also want it to overlap with the line:

Something like this would be sufficient:
```diff
+[data-contao--limit-height-target] {
+	position: relative;
+}

[data-contao--limit-height-target][style*="max-height"] .limit_toggler {
	background: linear-gradient(transparent, var(--content-bg));
}

.limit_toggler {
	width: 100%;
	position: absolute;
-	bottom: 0;
+	bottom: -1px;
	left: 0;
	line-height: 11px;
	text-align: center;
	z-index: 1;
}

![image](https://github.com/user-attachments/assets/05f2e024-aade-4f17-9d03-51cdae04d2f4)
![image](https://github.com/user-attachments/assets/b379f487-755c-4439-9c4e-02565601359d)

File: core-bundle/contao/themes/flexible/styles/main.css [hunk 1]
```diff
@@ -1656,20 +1656,24 @@ table.tl_listing {
...
 }
 
+[data-contao--limit-height-target][style*="max-height"] .limit_toggler {
+    background: linear-gradient(transparent, var(--content-bg));
+}
+
 .limit_toggler {
 	width: 100%;
 	position: absolute;
 	bottom: 0;
```
Comment (zoglo) [inline core-bundle/contao/themes/flexible/styles/main.css]: @lukasbableck there is one more issue that needs to be fixed

image
The toggler itself needs to be relative to the parent, however you also want it to overlap with the line:

```suggestion
[data-contao--limit-height-target] {
	position: relative;
}

[data-contao--limit-height-target][style*="max-height"] .limit_toggler {
	background: linear-gradient(transparent, var(--content-bg));
}

.limit_toggler {
	width: 100%;
	position: absolute;
	bottom: -1px;
```

Will look like that:

image
image

**Review by zoglo — CHANGES_REQUESTED** [review]

**Comment by zoglo** [comment]
Do you also want to resolve the conflict :)?

**Comment by lukasbableck** [comment]
Done 🙂

**Review by zoglo — DISMISSED** [review]

File: core-bundle/contao/themes/flexible/styles/main.css [hunk 2]
```diff
@@ -1656,21 +1656,35 @@ table.tl_listing {
...
+
+[data-contao--limit-height-target][style*="max-height"]:hover .limit_toggler {
+	background: linear-gradient(transparent, var(--hover-row) 60%);
+}
+
 .limit_toggler {
 	width: 100%;
 	position: absolute;
-	bottom: 0;
+	bottom: -1px;
```
Comment (leofeyer) [inline core-bundle/contao/themes/flexible/styles/main.css]:
```suggestion
	bottom: 0;
```

File: core-bundle/contao/themes/flexible/styles/main.css [hunk 3]
```diff
@@ -1656,21 +1656,35 @@ table.tl_listing {
...
+	display: flex;
+	justify-content: center;
+	align-items: end;
 }
 
 .limit_toggler button {
 	margin: 0;
 	padding: 0;
-	border: 0;
+	border: 1px solid var(--border);
```
Comment (leofeyer) [inline core-bundle/contao/themes/flexible/styles/main.css]:
```suggestion
	border: 0;
```

**Review by leofeyer — COMMENTED** [review]
Almost perfect, but I would prefer not to add a border around the toggle. It puts too much emphasize on the element IMHO.

**Review by zoglo — DISMISSED** [review]

**Review by leofeyer — APPROVED** [review]

**Comment by leofeyer** [comment]
Thank you @lukasbableck.

**Pull request merged**